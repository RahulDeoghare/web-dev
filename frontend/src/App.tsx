import { useEffect, useState } from 'react';
import './App.css';

function App() {
  const [message, setMessage] = useState('');
  const [name, setName] = useState('');
  const [response, setResponse] = useState('');

  // Use an environment variable or fallback to localhost
  const API_URL = process.env.REACT_APP_API_URL || 'http://192.168.29.119:5000';

  useEffect(() => {
    fetch(`${API_URL}/names`)
      .then((res) => {
        if (!res.ok) {
          throw new Error(`HTTP error! status: ${res.status}`);
        }
        return res.json();
      })
      .then((data) => setMessage(JSON.stringify(data)))
      .catch((err) => console.error('Fetch error:', err));
  }, [API_URL]);

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();

    if (!name.trim()) {
      setResponse('Name is required.');
      return;
    }

    try {
      const res = await fetch(`${API_URL}/name`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ name }),
      });

      if (!res.ok) {
        throw new Error(`HTTP error! status: ${res.status}`);
      }

      const data = await res.json();
      setResponse(data.message);
    } catch (err) {
      console.error('Fetch error:', err);
      setResponse('Failed to submit name. Please try again.');
    }
  };

  return (
    <div className="App">
      <h1>Enter Your Name</h1>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={name}
          onChange={(e) => setName(e.target.value)}
          placeholder="Your name"
        />
        <button type="submit">Submit</button>
      </form>
      <p>{response}</p>
      <h2>All Names:</h2>
      <pre>{message || 'Loading...'}</pre>
    </div>
  );
}

export default App;