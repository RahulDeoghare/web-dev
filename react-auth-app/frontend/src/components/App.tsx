import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Login from './Login';
import Register from './Register';

function App() {
  return (
    <Router>
      <div className="App">
        <Switch>
          <Route path="/login" component={Login} />
          <Route path="/register" component={Register} />
          <Route path="/" exact>
            <h1>Welcome to the Auth App</h1>
          </Route>
        </Switch>
      </div>
    </Router>
  );
}

export default App;