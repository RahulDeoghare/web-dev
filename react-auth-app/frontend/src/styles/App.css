body {
  font-family: Arial, sans-serif;
  background-color: #f4f4f4;
  margin: 0;
  padding: 0;
}

.App {
  text-align: center;
  padding: 20px;
}

h1 {
  color: #333;
}

form {
  background: #fff;
  padding: 20px;
  border-radius: 5px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  margin: 0 auto;
  max-width: 400px;
}

input[type="text"],
input[type="password"] {
  width: 100%;
  padding: 10px;
  margin: 10px 0;
  border: 1px solid #ccc;
  border-radius: 4px;
}

button {
  background-color: #5cb85c;
  color: white;
  padding: 10px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

button:hover {
  background-color: #4cae4c;
}

.error {
  color: red;
  margin-top: 10px;
}