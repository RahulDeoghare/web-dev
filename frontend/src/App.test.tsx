import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import App from './App';

// Always mock fetch to return a Promise with the expected shape
beforeEach(() => {
  (global.fetch as jest.Mock) = jest.fn(() =>
    Promise.resolve({
      ok: true,
      json: () => Promise.resolve([]),
    })
  );
});

afterEach(() => {
  jest.resetAllMocks();
});

// Suppress console.error messages in tests
beforeAll(() => {
  jest.spyOn(console, 'error').mockImplementation(() => {});
});
afterAll(() => {
  (console.error as jest.Mock).mockRestore();
});

// Test for heading
test('renders Enter Your Name heading', async () => {
  render(<App />);
  await waitFor(() => {
    expect(screen.getByText(/Enter Your Name/i)).toBeInTheDocument();
  });
});

// Test for input field
test('renders input field for name', async () => {
  render(<App />);
  await waitFor(() => {
    expect(screen.getByPlaceholderText(/Your name/i)).toBeInTheDocument();
  });
});

// Test for submit button
test('renders submit button', async () => {
  render(<App />);
  await waitFor(() => {
    expect(screen.getByText(/Submit/i)).toBeInTheDocument();
  });
});

// Test form submission (mock fetch)
test('submits the form and displays response', async () => {
  (global.fetch as jest.Mock)
    .mockResolvedValueOnce({
      ok: true,
      json: async () => ({ message: 'Name submitted!' }),
    })
    .mockResolvedValueOnce({
      ok: true,
      json: async () => ([]),
    });

  render(<App />);
  const inputElement = screen.getByPlaceholderText(/Your name/i);
  const buttonElement = screen.getByText(/Submit/i);

  fireEvent.change(inputElement, { target: { value: 'Rahul' } });
  fireEvent.click(buttonElement);

  await waitFor(() => {
    expect(screen.getByText(/Name submitted!/i)).toBeInTheDocument();
  });
});

// Test for loading state (if you display "Loading..." or similar)
test('shows loading state while fetching names', async () => {
  render(<App />);
  expect(screen.getByText(/Loading/i)).toBeInTheDocument();
  await waitFor(() => {
    // Wait for fetch to resolve and loading to disappear
    expect(screen.queryByText(/Loading/i)).not.toBeInTheDocument();
  });
});

// Test for displaying names fetched from API
test('displays names fetched from API', async () => {
  (global.fetch as jest.Mock).mockResolvedValueOnce({
    ok: true,
    json: async () => (["Alice", "Bob"]),
  });
  render(<App />);
  await waitFor(() => {
    expect(screen.getByText(/Alice/)).toBeInTheDocument();
    expect(screen.getByText(/Bob/)).toBeInTheDocument();
  });
});

// Test for API error handling
test('handles API error gracefully', async () => {
  (global.fetch as jest.Mock).mockResolvedValueOnce({
    ok: false,
    status: 500,
    json: async () => ({}),
  });
  render(<App />);
  // You can check for error logs or error UI if you display one
  // For now, just ensure the component doesn't crash
  await waitFor(() => {
    expect(screen.getByText(/Enter Your Name/i)).toBeInTheDocument();
  });
});

// Test for submitting empty input
test('does not submit when input is empty', async () => {
  render(<App />);
  const buttonElement = screen.getByText(/Submit/i);
  fireEvent.click(buttonElement);
  // Optionally, check for validation message or that fetch was not called
  expect(global.fetch).toHaveBeenCalledTimes(1); // Only the initial fetch
});

// Test for snapshot
test('matches the App component snapshot', async () => {
  const { asFragment } = render(<App />);
  // Wait for async effects to finish
  await waitFor(() => {
    expect(screen.getByText(/Enter Your Name/i)).toBeInTheDocument();
  });
  expect(asFragment()).toMatchSnapshot();
});
