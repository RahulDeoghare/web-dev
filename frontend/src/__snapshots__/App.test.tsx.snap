// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the App component snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Enter Your Name
    </h1>
    <form>
      <input
        placeholder="Your name"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <p />
    <h2>
      All Names:
    </h2>
    <pre>
      []
    </pre>
  </div>
</DocumentFragment>
`;
